source/common/network/dns_impl.cc

```diff
@@ -23,10 +23,10 @@ namespace Network {
 DnsResolverImpl::DnsResolverImpl(
     Event::Dispatcher& dispatcher,
     const std::vector<Network::Address::InstanceConstSharedPtr>& resolvers,
-    const bool use_tcp_for_dns_lookups)
+    const envoy::config::core::v3::DnsResolverOptions& dns_resolver_options)
     : dispatcher_(dispatcher),
       timer_(dispatcher.createTimer([this] { onEventCallback(ARES_SOCKET_BAD, 0); })),
-      use_tcp_for_dns_lookups_(use_tcp_for_dns_lookups),
+      dns_resolver_options_(dns_resolver_options),
```

Comment:
should this be instead 
```dns_resolver_options_(envoy::config::core::v3::DnsResolverOptions(dns_resolver_options)),``` ?
Since here we are storing the reference, we don't want this reference to point to nothing when original reference gets cleared up.

Reply:
We talked about this on slack, no?